### PR TITLE
feature: create login page

### PR DESCRIPTION
**Pull request by GuilhermeLLS**
### What is this
- This PR creates the login page and logic for the template.
- Add tests and stories for some componentes/pages.
- Fix miragejs server error

### How to test
You can run the app and mess around with the login form, run the tests and the stories.

### Observations
I had to force babel-loader version in template.json because storybook and react-scripts were in conflict due to babel-loader version.

### CODE REVIEW

**Comment by phenrique7** [comment]
> 1. Should I test components such as `Input` and `LogginWrapper`? They do not have any functionality at all, they are just pieces of UI.

No.

> 2. Are we going to have any UI sketchs to be used as guides? My UI skills are really bad hehe

For inspiration: https://tailwindui.com/components/application-ui/forms/sign-in-forms.

File: template/src/components/ControlledInput/index.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  InputRightElement,
+  Input,
+  InputProps,
+  FormLabel,
+  FormControl,
+  FormErrorMessage,
+} from '@chakra-ui/react';
+import { UseFormRegisterReturn } from 'react-hook-form';
+
+interface ControlledInputProps extends InputProps {
```
Comment (phenrique7) [inline template/src/components/ControlledInput/index.tsx:13]: We are already using `type` for typing, so let's standardize it.

File: template/src/components/ControlledInput/index.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  label: string;
+  icon?: React.ReactNode;
+  errorMessage?: string;
+  children?: React.ReactNode;
+  register?: UseFormRegisterReturn;
+}
+
+export default function ControlledInput(props: ControlledInputProps) {
```
Comment (phenrique7) [inline template/src/components/ControlledInput/index.tsx:21]: This component name does not seem appropriate to me. I suggest using the name `Input` indeed, adding an alias (`Input` as `ChakraInput`) for the Chakra UI's `Input` component in order to resolve the name conflict.

File: template/src/pages/Login/index.tsx [hunk 3]
```diff
@@ -1,16 +1,146 @@
...
+    mode: 'onChange',
+  });
+  const [isPasswordVisible, setIsPasswordVisible] = React.useState(false);
+  const [errorSubmitMessage, setErrorSubmitMessage] = React.useState('');
   const { authenticate } = useAuth();
 
+  const submitAction = ({ email, password }: FormInputs) => {
+    axios
+      .post('/api/login', { username: email, password })
+      .then(({ data: { token } }: AxiosResponse<any>) => {
```
Comment (diogomqbm) [inline template/src/pages/Login/index.tsx:40]: Shouldn't this type be something like `AxiosResponse<{ token: string}>`?

File: template/src/components/ControlledInput/index.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  return (
+    <FormControl isInvalid={!!errorMessage}>
+      <InputGroup flexDirection="column">
+        <FormLabel>{label}</FormLabel>
+        <Input {...register} {...remainingProps}></Input>
+        {icon && (
+          <InputRightElement top={[8, '2rem']} right={2}>
+            {icon}
+          </InputRightElement>
+        )}
```
Comment (phenrique7) [inline template/src/components/ControlledInput/index.tsx:40]:
```suggestion
        {icon ? (
          <InputRightElement top={[8, '2rem']} right={2}>
            {icon}
          </InputRightElement>
        ) : null}
```
Reasons for this change:
- https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx
- https://github.com/t10d/apolo-front/pull/8#discussion_r556556667

File: template/src/pages/Login/index.tsx [hunk 5]
```diff
@@ -1,16 +1,146 @@
...
+          >
+            Login
+          </Button>
+          {errorSubmitMessage.length > 0 && (
+            <Box mt="3" textAlign="center">
+              <Text color="red.500" fontWeight="semibold">
+                {errorSubmitMessage}
+              </Text>
+            </Box>
+          )}
```
Comment (phenrique7) [inline template/src/pages/Login/index.tsx:141]:
```suggestion
          {errorSubmitMessage.length > 0 ? (
            <Box mt="3" textAlign="center">
              <Text color="red.500" fontWeight="semibold">
                {errorSubmitMessage}
              </Text>
            </Box>
          ) : null}
```
https://github.com/t10d/cra-template-kingdom/pull/20#discussion_r614090996

File: template/src/components/LoginWrapper/index.tsx [hunk 6]
```diff
@@ -0,0 +1,31 @@
+import * as React from 'react';
+import { Flex, Heading, Link, Text } from '@chakra-ui/layout';
+
+interface LoginWrapperProps {
```
Comment (phenrique7) [inline template/src/components/LoginWrapper/index.tsx:4]: https://github.com/t10d/cra-template-kingdom/pull/20#discussion_r614083723

**Comment by GuilhermeLLS** [comment]
@phenrique7 @diogomqbm done!

**Review by phenrique7 — APPROVED** [review]

**Pull request merged**